### PR TITLE
replace PE with compiler(new feature in paddle151).

### PR DESCRIPTION
**Pull request by TomorrowIsAnOtherDay**
### This PR:
- add `parl.compile`.
- replace PE with compiler in A2C/IMPALA. 

```python
learn_program = fluid.Program()
....define the learn_program and cost 

learn_program = parl.compile(learn_program, cost.name)
agent.fluid_executor.run(learn_program)
```

I haven't used this API for GA3C/competition directory as it may cause thread-safe problems when using `parl.executor` in different threads.

### CODE REVIEW

**Review by zenghsh3 — CHANGES_REQUESTED** [review]
paddle dependence in homepage, examples/A2C, examples/IMPALA need update.

File: examples/IMPALA/atari_agent.py [hunk 1]
```diff
@@ -86,8 +73,8 @@ def build_program(self):
             vtrace_loss, kl = self.alg.learn(obs, actions, behaviour_logits,
                                              rewards, dones, lr, entropy_coeff)
             self.learn_outputs = [
-                vtrace_loss.total_loss.name, vtrace_loss.pi_loss.name,
-                vtrace_loss.vf_loss.name, vtrace_loss.entropy.name, kl.name
+                vtrace_loss.total_loss, vtrace_loss.pi_loss,
+                vtrace_loss.vf_loss, vtrace_loss.entropy, kl
```
Comment (zenghsh3) [inline examples/IMPALA/atari_agent.py:77]: learn program is not compiled here.

**Review by zenghsh3 — CHANGES_REQUESTED** [review]

File: examples/IMPALA/atari_agent.py (hunk 1 above)
Comment (TomorrowIsAnOtherDay) [reply examples/IMPALA/atari_agent.py:77]: done

**Review by zenghsh3 — APPROVED** [review]
LGTM

**Pull request merged**